### PR TITLE
 create an asciicast

### PR DESCRIPTION
**Pull request by derhuerst**
Created with `npm run asciicast`. #8

### CODE REVIEW

File: example-all.js [hunk 1]
```diff
@@ -2,20 +2,26 @@
...
-setInterval(function () {
-	if (++i === frames.length) {
-		j++;
-		i = 0;
+var showNextSpinner = function () {
+	if (next) {
+		clearInterval(next);
+		spinner++;
 	}
+	if (spinner in spinners) {
```
Comment (sindresorhus) [inline example-all.js]: Would be better to just check the array length against the index.

File: example-all.js [hunk 2]
```diff
@@ -2,20 +2,26 @@
...
+		var s = cliSpinners[spinners[spinner]];
+		next = setInterval(showNextFrame, s.interval || 100);
+		setTimeout(showNextSpinner, Math.max((s.interval || 100) * s.frames.length, 1000));
```
Comment (sindresorhus) [inline example-all.js]: No need for the `100` fallbacks. All spinners have `interval` defined.

File: package.json [hunk 3]
```diff
@@ -13,7 +13,8 @@
     "node": ">=0.10.0"
   },
   "scripts": {
-    "test": "xo && ava"
+    "test": "xo && ava",
+    "asciicast": "asciinema rec -c 'node example-all.js' -t 'cli-spinner' -q"
```
Comment (sindresorhus) [inline package.json]: Can you use long flags here?

File: readme.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 > 50+ spinners for use in the terminal
 
-![](screenshot.gif)
+[![asciicast](https://asciinema.org/a/38570.png)](https://asciinema.org/a/38570)
```
Comment (sindresorhus) [inline readme.md]: I would still like to show the gif somewhere in the readme for the lazy people that don't bother to click.

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by sindresorhus** [comment]
Look at the diff. The conflict was incorrectly resolved ;)

Also: https://github.com/sindresorhus/cli-spinners/pull/10#discussion_r55322402

**Comment by derhuerst** [comment]
> Look at the diff. The conflict was incorrectly resolved ;)

In which way? Sorry, but I don't get it. :smile:

**Comment by sindresorhus** [comment]
You've included changes that are already commits in master. For example: https://github.com/sindresorhus/cli-spinners/pull/10/files#diff-7cd1ff39b4826900412165bf40ac99a7R280

**Comment by sindresorhus** [comment]
ping :)

**Pull request closed without merging**